particle-services/src/app_services.rs

```diff
@@ -482,15 +484,20 @@ impl ParticleAppServices {
         //     ));
         // }
         // Metrics collection are enables for services with aliases which are installed on root worker or worker spells.
-        let service_type = self.get_service_type(service, &worker_id);
+        let service_type = self.get_service_type(service.as_ref(), &peer_scope);
 
         // TODO: move particle vault creation to aquamarine::particle_functions
         if create_vault {
             self.vault.create(&particle.id)?;
         }
 
+        let call_parameters_host_id = match peer_scope {
+            PeerScope::WorkerId(worker_id) => worker_id.to_string(),
+            PeerScope::Host => self.scope.get_host_peer_id().to_string(),
+        };
```

Comment:
Maybe we should extend this PeerScope fellow to something like `scope.to_peer_id_or(host_id)` or smth with more conventional naming to reduce the number of patter-matching?   
```suggestion
        let call_parameters_host_id = peer_scope.to_peer_id_or(self.scope.get_host_peer_id().to_string()),
```

This would also improve that ugly interface for `CallParams::local`